Artifacthub branch update fails on massive policies updates

When all the policies are being released at the same time, the CI job that updates the artifacthub branch can fail:

```
+ git config user.name 'Update artifacthub branch'
+ git config user.email github-actions@github.com
+ git checkout -b main --track origin/main
Previous HEAD position was e24b1bf79 Merge pull request #315 from kubewarden/updatecli_main_policies/cel-policy
Switched to a new branch 'main'
branch 'main' set up to track 'origin/main'.
+ git checkout -b artifacthub --track origin/artifacthub
Switched to a new branch 'artifacthub'
branch 'artifacthub' set up to track 'origin/artifacthub'.
+ git reset HEAD -- .
+ git checkout cel-policy/v1.5.6 -- artifacthub-repo.yml
+ mkdir -p policies/cel-policy
+ mv /home/runner/work/_temp/artifacthub-pkg.yml policies/cel-policy/artifacthub-pkg.yml
+ git checkout cel-policy/v1.5.6 -- policies/cel-policy/README.md
+ git add artifacthub-repo.yml
+ git add policies/cel-policy/artifacthub-pkg.yml
+ git add policies/cel-policy/README.md
++ sed --posix -n 's,^version: \(.*\),\1,p' policies/cel-policy/artifacthub-pkg.yml
+ VERSION=1.5.6
+ git commit -m 'Bump ArtifactHub files for policies/cel-policy, version 1.5.6'
[artifacthub ad27f85ad] Bump ArtifactHub files for policies/cel-policy, version 1.5.6
 1 file changed, 17 insertions(+), 17 deletions(-)
+ for i in {1..5}
+ git pull --ff-only origin artifacthub
From https://github.com/kubewarden/policies
 * branch                artifacthub -> FETCH_HEAD
   4b213b5ce..765be7e7d  artifacthub -> origin/artifacthub
hint: Diverging branches can't be fast-forwarded, you need to either:
hint:
hint: 	git merge --no-ff
hint:
hint: or:
hint:
hint: 	git rebase
hint:
hint: Disable this message with "git config set advice.diverging false"
fatal: Not possible to fast-forward, aborting.
Error: Process completed with exit code 128.
```

A full example of a failure can be found [here](https://github.com/kubewarden/policies/actions/runs/21398474361/job/61603507669#logs), in the fist run of the CI. 

As far as I can see, this is caused by the concurrent updates in the branch. Therefore, we need to improve the retry mechanism already in place or find a way to make updates sequential. 

### Acceptance criteria

- Find the best solution for this issue
- Test a massive policy updates

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -265,6 +265,9 @@ jobs:
       # added or changed files to the repository.
       contents: write
     runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-artifacthub
+      cancel-in-progress: false
     steps:
       - name: Checkout code
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
@@ -324,8 +327,9 @@ jobs:
           VERSION=$(sed --posix -n 's,^version: \(.*\),\1,p' ${{ needs.calculate-policy-from-tag.outputs.policy-working-dir }}/artifacthub-pkg.yml)
           git commit -m "Bump ArtifactHub files for ${{ needs.calculate-policy-from-tag.outputs.policy-working-dir }}, version $VERSION"
 
-          # push can fail as several release jobs run simultaneuously and there's new changes.
-          # Retry in loop with rebase to replay our commit on top of any concurrent pushes:
+          # The ArtifactHub branch is updated by many release runs. The job
+          # concurrency group serializes normal releases; keep the rebase retry
+          # for any out-of-band branch update that lands while this job runs.
           for i in {1..5}; do
             git pull --rebase origin artifacthub
             if git push origin artifacthub; then

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -266,7 +266,7 @@ jobs:
       contents: write
     runs-on: ubuntu-latest
     concurrency:
-      group: ${{ github.workflow }}-artifacthub
+      group: artifacthub-branch-updates
       cancel-in-progress: false
     steps:
       - name: Checkout code

